src/main/java/io/appform/functionmetrics/FunctionTimerAspect.java

```diff
@@ -43,8 +50,34 @@ public Object around(ProceedingJoinPoint joinPoint) throws Throwable {
         final String methodName = Strings.isNullOrEmpty(monitoredFunction.method())
                                     ? callSignature.getName()
                                     : monitoredFunction.method();
-        log.trace("Called for class: {} method: {}", className, methodName);
-        final FunctionInvocation invocation = new FunctionInvocation(className, methodName);
+        final String[] parameterNames = methodSignature.getParameterNames();
+        final Parameter[] parameters = methodSignature.getMethod().getParameters();
+        String parameterString = "";
+        if (parameterNames != null && parameterNames.length > 0) {
+            List<String> dynamicPrefixComponents = IntStream.range(0, parameterNames.length)
+                    .mapToObj(i -> {
+                        String parameterName = parameterNames[i];
+                        Object[] args = joinPoint.getArgs();
+                        if (args != null && i < args.length) {
```

Comment:
Move this check outside. How is the second condition possible? Do a length check on the array before getting into the loop.